app/Http/Controllers/AuthorController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\AuthorRequest;
+use App\Repositories\RepositoryInterface\BaseRepositoryInterface;
+
+class AuthorController extends Controller
+{
+    protected $authorRepository;
+
+    public function __construct(BaseRepositoryInterface $authorRepository){
+        $this->authorRepository = $authorRepository;
+    }
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */ 
+    public function index()
+    {
+        $author = $this->authorRepository->getAuthor();
+
+        return view('authors.view', compact('author'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('authors.add');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(AuthorRequest $request)
+    {
+        if ($this->authorRepository->createAuthor($request->only('author_name', 'author_avatar', 'author_desc'))) {
+            return redirect()->route('admin.authors.index')->with('add_success', trans('message.author.add_success'));
+        }
+
+        return redirect()->route('admin.authors.index')->with('Fail', trans('message.author.add_fail'));
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        //
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($author_id)
```

Comment:
```suggestion
    public function edit($authorId)
```
sửa lại hết tên biến nhé em

Reply:
dạ :(